src/structure.rs

```diff
@@ -16,8 +18,9 @@ pub const BASE_SUBPATH: &str = "pkgs/by-name";
 pub const PACKAGE_NIX_FILENAME: &str = "package.nix";
 
 lazy_static! {
-    static ref SHARD_NAME_REGEX: Regex = Regex::new(r"^[a-z0-9_-]{1,2}$").unwrap();
-    static ref PACKAGE_NAME_REGEX: Regex = Regex::new(r"^[a-zA-Z0-9_-]+$").unwrap();
+    static ref SHARD_NAME_REGEX: Regex = Regex::new(r"^((_[0-9])|([a-z][a-z0-9_-]?))$").unwrap();
+    static ref PACKAGE_NAME_REGEX: Regex =
+        Regex::new(r"^((_[0-9])|[a-zA-Z])[a-zA-Z0-9_-]*$").unwrap();
```

Comment:
```suggestion
    static ref SHARD_NAME_REGEX: Regex = Regex::new(r"^[a-z_-][a-z0-9_-]?$").unwrap();
    static ref PACKAGE_NAME_REGEX: Regex = Regex::new(r"^[a-zA-Z_-][a-zA-Z0-9_-]*$").unwrap();
```

I think this is a bit simpler to understand

Reply:
We actually don't even need a new error type for this, because we can just update the message for when these regexes don't match here: https://github.com/NixOS/nixpkgs-vet/blob/02db8e474807270e5611182141f764722345d319/src/problem/npv_110.rs#L19 and https://github.com/NixOS/nixpkgs-vet/blob/02db8e474807270e5611182141f764722345d319/src/problem/npv_141.rs#L22

Reply:
> static ref PACKAGE_NAME_REGEX: Regex = Regex::new(r"^[a-zA-Z_-][a-zA-Z0-9_-]*$").unwrap();

This allows `_abc` and even `___` as package name.